### PR TITLE
[#2279] improvement(spark): Trigger the upstream rewrite when the read stage fails

### PR DESCRIPTION
**Pull request by yl09099**
### What changes were proposed in this pull request?

If the current Reader fails to obtain Shuffle data, it does not trigger the upstream Stage to rewrite the data. If a Shuffle Server fails, it does not trigger Stage retry.

### Why are the changes needed?

Fix: #2279 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

UT.

### CODE REVIEW

**Comment by jerqi** [comment]
Could you add a ut for this?

File: client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java [hunk 1]
```diff
@@ -466,16 +490,36 @@ public void incPartitionFetchFailure(int stageAttempt, int partition) {
...
+              if (this.partitions[partition] >= shuffleManager.getMaxFetchFailures()) {
+                return true;
+              } else {
+                return false;
+              }
             }
           });
     }
+
+    public void setClearedMapTrackerBlock(boolean isCleared) {
```
Comment (jerqi) [inline client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java:509]: Could you give a better name for this?

File: client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java [hunk 2]
```diff
@@ -162,18 +164,37 @@ public void reportShuffleFetchFailure(
         code = RssProtos.StatusCode.INVALID_REQUEST;
         reSubmitWholeStage = false;
       } else { // update the stage partition fetch failure count
-        code = RssProtos.StatusCode.SUCCESS;
-        status.incPartitionFetchFailure(stageAttempt, partitionId);
-        int fetchFailureNum = status.getPartitionFetchFailureNum(stageAttempt, partitionId);
-        if (fetchFailureNum >= shuffleManager.getMaxFetchFailures()) {
```
Comment (zuston) [inline client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java:168]: Why removing this logic?

**Comment by yl09099** [comment]
I'm not ready yet. Wait a minute

**Pull request closed without merging**